primitives/io/src/lib.rs

```diff
@@ -351,6 +351,19 @@ pub trait Misc {
 	fn print_hex(data: &[u8]) {
 		log::debug!(target: "runtime", "{}", HexDisplay::from(&data));
 	}
+
+	/// Extract the runtime version of the given wasm blob by calling `Core_version`.
+	///
+	/// Returns the SCALE encoded runtime version and `None` if the call failed.
+	fn runtime_version(&mut self, wasm: &[u8]) -> Option<Vec<u8>> {
+		// Create some dummy externalities, `Core_version` should not write data anyway.
```

Comment:
What kind of side effects? Even if the call would write something, it would not leak into the Externalities of the actual block. If that was your question.

Reply:
Yes, that's what I meant, I wanted to be sure that the two contexts do not share anything.